### ISSUE TITLE
Editor crash when duplicating node

### ISSUE THREAD

**Issue by endragor** [godotengine/godot#13432]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
macOS 10.12.6, Godot 9678231b109c333a5273325c8758241310cd27f4

**Issue description:**

Editor may crash when duplicating node (stack trace below). This happens stably for some nodes, not sure which "special" properties those nodes posses though.

```
ERROR: get_child: Index p_index=2 out of size (data.children.size()=2)
   At: scene/main/node.cpp:1434.
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fffc71b7b3a _sigtramp + 26
[2] 2   ???                                 0x00007fff5639dbf8 0x0 + 140734640020472
[3] Node::duplicate_from_editor(Map<Node const*, Node*, Comparator<Node const*>, DefaultAllocator>&) const (in godot.osx.tools.64) (node.cpp:2223)
[4] SceneTreeDock::_tool_selected(int, bool) (in godot.osx.tools.64) (scene_tree_dock.cpp:448)
[5] SceneTreeDock::_unhandled_key_input(Ref<InputEvent>) (in godot.osx.tools.64) (scene_tree_dock.cpp:80)
[6] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[7] Object::call_multilevel(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:747)
[8] SceneTree::_call_input_pause(StringName const&, StringName const&, Ref<InputEvent> const&) (in godot.osx.tools.64) (scene_tree.cpp:878)
[9] Viewport::unhandled_input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:2423)
[10] Viewport::_vp_unhandled_input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:1331)
[11] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[12] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:913)
[13] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:838)
[14] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (scene_tree.cpp:246)
[15] SceneTree::input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (scene_tree.cpp:420)
[16] InputDefault::parse_input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (input_default.cpp:345)
[17] OS_OSX::push_input(Ref<InputEvent> const&) (in godot.osx.tools.64) (os_osx.mm:1969)
[18] -[GodotContentView keyDown:] (in godot.osx.tools.64) (os_osx.mm:775)
[19] 19  AppKit                              0x00007fffafc102cc -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 4086
[20] 20  AppKit                              0x00007fffafc0ef0a -[NSWindow(NSEventRouting) sendEvent:] + 541
[21] 21  AppKit                              0x00007fffafa944a8 -[NSApplication(NSEvent) sendEvent:] + 4768
[22] -[GodotApplication sendEvent:] (in godot.osx.tools.64) (os_osx.mm:110)
[23] OS_OSX::process_events() (in godot.osx.tools.64) (os_osx.mm:1949)
[24] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:1994)
[25] main (in godot.osx.tools.64) (godot_main_osx.mm:90)
[26] 26  libdyld.dylib                       0x00007fffc6fa8235 start + 1
-- END OF BACKTRACE --
```

**Comment by akien-mga** [comment]
CC @RandomShaper

**Comment by endragor** [comment]
Also happens when duplicating nodes via scripting API (since the issue is in `duplicate` method). @kosz78 is preparing a fix.

**Comment by akien-mga** [comment]
Fixed by #13445.
